front/app/components/admin/GraphCards/CommentsByTimeCard/useCommentsByTime/parse.ts

```diff
@@ -0,0 +1,123 @@
+import moment, { Moment } from 'moment';
+
+// utils
+import { timeSeriesParser } from 'components/admin/GraphCards/_utils/timeSeries';
+import { get } from 'utils/helperUtils';
+import { orderBy } from 'lodash-es';
+
+// typings
+import {
+  Response,
+  TimeSeriesResponseRow,
+  TimeSeries,
+  TimeSeriesRow,
+} from './typings';
+import { Translations } from './translations';
+import { IResolution } from 'components/admin/ResolutionControl';
+
+export const getEmptyRow = (date: Moment) => ({
+  date: date.format('YYYY-MM-DD'),
+  comments: 0,
+  total: 0,
+});
+
+const parseRow = (date: Moment, row?: TimeSeriesResponseRow): TimeSeriesRow => {
+  if (!row) return getEmptyRow(date);
+
+  return {
+    total: 0,
+    comments: row.count,
+    date: date.format('YYYY-MM-DD'),
+  };
+};
+
+const getDate = (row: TimeSeriesResponseRow) => {
+  return moment(get(row, 'first_dimension_date_created_date'));
+};
+
+const _parseTimeSeries = timeSeriesParser(getDate, parseRow);
+
+export const parseTimeSeries = (
```

Comment:
Maybe we should turn this into a reusable function somehow, since it's doing the exact same thing in three different places... a `calculateTotal` utility or something?

Reply:
its not exact since the total is coming from different types of objects, the substraction could be done with one or multiple values and the variable names are different. got to move along with this PR but glad to pair and try to build a reusable function using reduce as well.